include/bit_algorithm_details.hpp

```diff
@@ -1,29 +1,96 @@
-// ======================== BIT_ALGORITHM_DETAILS =========================== //
-// Project:     The Experimental Bit Algorithms Library
-// Name:        iterator_utils.hpp
-// Description: Implementation of algorithm-related utilities not present in 
-// 		        algorithms header.
-// Creator:     Vincent Reverdy
-// Contributor: Bryce Kille [2019]
-//              Collin Gress [2019]
-// License:     BSD 3-Clause License
+// ========================= BIT ALGORITHM DETAILS  ========================= //
+// Project: The Experimental Bit Algorithms Library
+// Name: bit_algorithm_details.hpp
+// Description: A set of utilities to assist in writing algorithms 
+// Creator: Vincent Reverdy
+// Contributor(s): Vincent Reverdy [2019]
+//                 Collin Gress [2019]
+//                 Bryce Kille [2019]
+// License: BSD 3-Clause License
 // ========================================================================== //
 #ifndef _BIT_ALGORITHM_DETAILS_HPP_INCLUDED
 #define _BIT_ALGORITHM_DETAILS_HPP_INCLUDED
 // ========================================================================== //
 
 
-
-// ================================ PREAMBLE ================================ //
+// ============================== PREAMBLE ================================== //
 // C++ standard library
 // Project sources
-#include "bit.hpp"
 // Third-party libraries
 // Miscellaneous
-namespace bit {
-// ========================================================================== //
 
+namespace bit {
+/*
+ * Takes two words and merges them together.
+ * 
+ * @param w1 the first operand of the merge
+ * @param w2 the second operand of the merge
+ * @param w1_position the position within w1 to start the merge from
+ * @return a word given by merging w1 and w2
+ */ 
+template <class T>
+T _shift_and_merge(T w1, T w2, std::size_t w1_position) {
```

Comment:
How is this different than `_shld`? It seems like the only difference is that this assumes that `_shift_and_merge` guarantees that `w1_position` is within the word and `_shld` does not. If this is the case, should we create an overload of `_shld` or something? It seems odd to have two functions with nearly identical functionality.